### DISCUSSION TITLE
PostFix as Smarthost for Mailserver @home

### DISCUSSION THREAD

**Discussion by TheBigM85** [postfixadmin/postfixadmin#791] [General]
Hi.

Is not a question to postfix Admin, but for postfix gerneral...

Dies ansinge can tell me, where i can find a how to or have Input for me, how to configure postfix, to relay mail from my homeserver (Mailcow) as smarthost to internet.
Postfix is on small Internet VPS with Static ip an Mailcow on Server @home with dynamic ip an MyFRITZ DDNS. I want to send mail from Mailcow over postfix on vps (authenticated with smtp Account or sasl). Unbound via MX to myfritz Adress. SPF to ip of VPS.

How to configure postfix on VPS to only accept inbound (send) from my Mailcow?

Thx

Mario

Sorry for Bad english :)

In my nativ language (german):

Hi.

Ich möchte zu hause an einem Anschluss einen privaten Mailserver mit meinen Domains betreiben (habe da auch schon ausreichend Erfahrung (setze hier Mailcow ein)).
Leider habe ich keine statische IP, was ausgehend ein Problem ist (bitte dazu keine Grundsatzdiskussionen, sondern bitte nur explizit zu meiner Frage! Danke).

Ich habe extern noch einen VPS mit statischer IP und nextcloud und meine Idee (und glaube zu wissen das geht), wäre hier Postfix als eigenen Smarthost einzusetzen, der als relay für meinen Server zu Hause fungiert und statt diesem die Mails direkt versendet.

Frage:
Wie konfiguriere ich hier den postfix per SMTP-Auth, dass dieser nur Mails von meinem Mailcowserver annimmt (über konfigurierten SMTP user; alternativ SASL (Mailcow möglich?)).
Hat da einer Erfahrung / Beispielconfig?

Danke im Voraus.

**Comment by TheBigM85** [comment]
Additional:
Is it easier to use Postfixadmin for this configuration?

How to Setup on VPS with NextCloud on this and postfix installed?

Or have i to use clean server for this?

Thx

**Comment by drdelaney** [comment]
I do not know if it can be done in postfixadmin, I still need to finalize my transition from qmail...
But as far as doing it from raw postix, it should be simple. I do not know if it can be easily done on a working server, as I started with a clean server.

You want to look into the option `relayhost`. Along with that, probably some TLS options.

This is what I have in my server which seems to be working:
```
relayhost = [mail.myhostname.com]:587
smtp_use_tls = yes
smtp_sasl_auth_enable = yes
smtp_sasl_password_maps = hash:/etc/postfix/sasl_passwd
smtp_sasl_security_options = noanonymous
smtp_sasl_tls_security_options = noanonymous
# openssl req -new -x509 -nodes -out s/etc/postfix/tls/mtp-client.pem -keyout s/etc/postfix/tls/mtp-client.pem -days 10000
smtp_tls_cert_file = /etc/postfix/tls/smtp-client.pem
smtp_tls_key_file = $smtp_tls_cert_file
tls_random_source = dev:/dev/urandom
smtp_tls_session_cache_database = btree:${data_directory}/smtp_scache
smtp_tls_loglevel = 2
smtp_tls_per_site = hash:/etc/postfix/tls/tls_sites
```

And then the supplemental files:
```
[root@nibbler postfix]# cat /etc/postfix/tls/tls_sites
mail.myhostname.com MUST
[root@nibbler postfix]# cat /etc/postfix/sasl_passwd
[mail.myhostname.com]:587 username@myhostname.com:PassWordHere
```

As always, your mileage may vary.

**Reply by TheBigM85** [reply]
Hi.

This is not, what i want. This configuration is to use postfix for relay to internet Mailserver.

I want to use postfix on smtp to send Mail Directly to Internet and accept incoming mail from my mailserver at home to send Mails outgoing through the postfix on the vps.

I think your configuration is to send mail through smarthost on external Provider Like Strato or so.

I Need / want Information how to configure postfix on Internet Site vps to accept Mails only from my Server @home (smtp auth. or sasl).

Thx in advance

Mario

Sorry if i missunderstand your answer.

Sorry for bad english and autocorrect errors 🫣😅

**Reply by drdelaney** [reply]
it was me who misunderstood you.  my example is indeed for sending mail to an outside host.

For handling mail, most likely set up postfixadmin as normal (probably on a fresh install).  It should be able to store mail locally for you as well as relay mail out to the internet.  You should be able to add in any domain, including "@home" if you want, but this might be against RFC so you may want to do something "@mail.home" or "@mail.local" if you do not have a valid domain name.
Keep in mind, it should have a valid hostname if you want to send email out to the internet.

Hopefully I understood this time.

**Reply by TheBigM85** [reply]
Hi. You misunderstood me again. I have a few domains and also access to the DNS and I already have experience with Linux, Postfix and also Mailcow etc. I have a server at home and now want to run Mailcow there, which is no problem (also for the receiving path with a dynamic IP). However, I need a static IP for the sending path, hence the VPS with Postfix, which will serve as a smart host for sending for my server at home. So I just need to know (unfortunately I don't know) how to set up Postfix on the VPS so that my mail server at home can authenticate itself to it (via SMTP-Auth or sasl) and then send its emails directly via it. 

Thanks in advance.

This time translatet with google translate 😉

**Reply by drdelaney** [reply]
I am misunderstanding again, because this sounds like my first reply would fit your situation.  Maybe post in your native language to explain it better and someone else can pick up.  Sorry I cannot help further myself.
